common4j/src/main/com/microsoft/identity/common/java/commands/InteractiveTokenCommand.java

```diff
@@ -79,7 +81,23 @@ public AcquireTokenResult execute() throws Exception {
 
                 span.setAttribute(AttributeName.controller_name.name(), controller.getClass().getSimpleName());
 
-                return controller.acquireToken((InteractiveTokenCommandParameters) getParameters());
+                final AcquireTokenResult result = controller.acquireToken((InteractiveTokenCommandParameters) getParameters());
+
+                if (result == null) {
+                    span.setStatus(StatusCode.ERROR, "empty result");
+                } else if (result.getSucceeded()) {
+                    span.setStatus(StatusCode.OK);
+                } else {
+                    final BaseException exception = ExceptionAdapter.exceptionFromAcquireTokenResult(result, getParameters());
+                    if (exception != null) {
+                        span.recordException(exception);
+                        span.setStatus(StatusCode.ERROR);
```

Comment:
I feel like setting statusCode to ERROR can be handled inside the recordException call, to avoid someone recording the exception without updating the status. Are there any scenarios where we record an exception but do not set the status to error?

Reply:
We don't own the `recordException` so we can't put code inside that method.